From 4735c6dd6b9413756f452959a6421e31e7d588fa Mon Sep 17 00:00:00 2001
From: Haim Boot <hayim@marvell.com>
Date: Sun, 24 Jul 2016 18:19:34 +0300
Subject: [PATCH 0378/1345] doc: thermal: add irq-chip entry in dt-bindings

commit  51937414b458531c00ff6c6ef953084672a29c74 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

add new entrym,interrupts-extended, which replaces
the interrupts entry when using irq-chip.

Change-Id: I3cfcc914c188f204108e3594d7ce2e8d87f5612f
Signed-off-by: Haim Boot <hayim@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/31391
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../devicetree/bindings/thermal/armada-thermal.txt |   23 ++++++++++++++++++--
 1 file changed, 21 insertions(+), 2 deletions(-)

diff --git a/Documentation/devicetree/bindings/thermal/armada-thermal.txt b/Documentation/devicetree/bindings/thermal/armada-thermal.txt
index 79ee568..c2cb8ee 100644
--- a/Documentation/devicetree/bindings/thermal/armada-thermal.txt
+++ b/Documentation/devicetree/bindings/thermal/armada-thermal.txt
@@ -19,6 +19,10 @@ Required properties:
 		Armada 38x.
 
 - interrupts:	Should contain interrupt for thermal system
+or
+- interrupts-extended:
+		Should contain parent node and interrupt for thermal system
+		(relevant for devices with irq-chip).
 
 - threshold:	Temperature threshold in Celsius, the threshold +/- calculated
 		hysteresis (formula below) define the upper and lower watermark
@@ -29,13 +33,28 @@ Required properties:
 		Following are the corresponding degrees corresponding to
 		hysteresis value: 0=>1.9, 1=>3.8,  2=>7.6, 3=>15.2
 
-Example:
+Example for a38x:
 
 	thermal@e4078 {
 		compatible = "marvell,armada380-thermal";
-		reg = <0xe4078 0x4>, <0xe4074 0x4>, <0xe4100 0x20>;
+		reg = <0xe4078 0x4>,
+		      <0xe4074 0x4>,
+		      <0xe4100 0x20>;
 		interrupts = <GIC_SPI 78 IRQ_TYPE_LEVEL_HIGH>;
 		threshold = <100>;
 		hysteresis = <2>;
 		status = "okay";
 	};
+
+or Example for Armada-7040 (with irq-chip):
+
+	thermal@400078 {
+		compatible = "marvell,armada-cp110-thermal";
+		reg = <0x400078 0x4>,
+		      <0x400070 0x8>,
+		      <0x400100 0x20>;
+		interrupts-extended = <&sei 37>;
+		threshold = <100>;
+		hysteresis = <2>;
+		status = "okay";
+	}
-- 
1.7.9.5

